src/Styleguide/Utils/Responsive.tsx

```diff
@@ -38,10 +45,15 @@ export class ResponsiveProvider extends React.Component<
       mediaMatchers: [],
       mqHandler: null,
     }
+    this.setupObservers(this.props.breakpoints)
   }
 
-  componentDidMount() {
-    this.setupObservers(this.props.breakpoints)
+  shouldComponentUpdate(nextProps, nextState) {
+    if (nextProps.children !== this.props.children) return true
+    if (shallowEqual(this.state.breakpoints, nextState.breakpoints)) {
+      return false
+    }
+    return true
```

Comment:
This requires some testing (especially because i'm not sure about shallow comparisons of children as a data structure), but maybe React.PureComponent would work well here.

Reply:
I agree that it'll require testing. This approach reduces the render duplicating over what `PureComponent` does though.